crates/subspace-farmer/src/farming.rs

```diff
@@ -37,7 +37,7 @@ pub enum FarmingError {
 /// in its `Commitments` database.
 pub struct Farming {
     stop_sender: async_oneshot::Sender<()>,
-    handle: Option<JoinHandle<Result<(), FarmingError>>>,
+    handle: Fuse<JoinHandle<Result<(), FarmingError>>>,
```

Comment:
Wow, never actually thought about `Fuse` as a smart wrapper for `Option`

Reply:
It is not, Fuse here is so that we can poll future after it has completed with panic.

Reply:
I updated comment here, it should have been "without". Fused future wrapper tracks completion and gives you `is_terminated()` method, so you can avoid calling future after completion and thus avoid panic and other bad side effects: https://doc.rust-lang.org/std/future/trait.Future.html#panics